src/littlefs/__main__.py

```diff
@@ -0,0 +1,134 @@
+            with fs.open(rel_path.as_posix(), "wb") as dest:
+                with open(path, "rb") as src:
+                    dest.write(src.read())
+    with open(args.image, "wb") as fp:
+        fp.write(fs.context.buffer)
+
+
+def _list(args: argparse.Namespace):
+    if args.block_size is None or args.block_count is None:
+        print("Block size and block count is required to create an image.")
+        return -1
+
+    fs = LittleFS(block_size=args.block_size, block_count=args.block_count)
+    with open(args.image, "rb") as fp:
+        fs.context.buffer = bytearray(fp.read())
+
+    fs.mount()
+    for root, dirs, files in fs.walk("/"):
+        if not root.endswith("/"):
+            root += "/"
+        for dir in dirs:
+            print(f"{root}{dir}")
+        for file in files:
+            print(f"{root}{file}")
+
+
+def _unpack(args: argparse.Namespace):
+    if args.block_size is None or args.block_count is None:
+        print("Block size and block count is required to create an image.")
+        return -1
+
+    fs = LittleFS(block_size=args.block_size, block_count=args.block_count)
+    with open(args.image, "rb") as fp:
+        fs.context.buffer = bytearray(fp.read())
+
+    root_dest: pathlib.Path = args.destination
+    root_dest = root_dest.absolute()
+    if not root_dest.exists() or not root_dest.is_dir():
+        print("Destination directory does not exist")
+
+    fs.mount()
+    for root, dirs, files in fs.walk("/"):
+        if not root.endswith("/"):
+            root += "/"
+        for dir in dirs:
+            src_path = root + dir
+            dst_path = root_dest / src_path[1:]
+            print(src_path, dst_path)
+            assert root_dest in dst_path.parents
+            dst_path.mkdir(exist_ok=True)
+        for file in files:
+            src_path = root + file
+            dst_path = root_dest / src_path[1:]
+            print(src_path, dst_path)
+            assert root_dest in dst_path.parents
+            with fs.open(src_path, "rb") as src:
+                dst_path.write_bytes(src.read())
+
+
+def main():
+    if sys.argv[0].endswith("__main__.py"):
+        prog = f"python -m littlefs"
+    else:
+        prog = f"python-littlefs"
+
+    parser = argparse.ArgumentParser(
+        prog=prog,
+        description=textwrap.dedent(
+            """\
+            Create, extract and inspect LittleFS filesystem images. Use one of the
+            commands listed below, the '-h' / '--help' option can be used on each
+            command to learn more about the usage.
+            """
+        ),
+        # formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    subparsers = parser.add_subparsers(required=True, title="available commands", dest="command")
+
+    parser_create = subparsers.add_parser(
+        "create", help="Create LittleFS image from directory content"
+    )
+    parser_create.add_argument("source", help="Source Directory", type=pathlib.Path)

```

Comment:
an alternative to manually defining the `meth` dictionary below, is that you can do something like this:

```
parser_create.set_defaults(func=_create)

# Do the same for the other commands

args.parse_args()

args.func(args)  # invoke the subcommand and pass in the args Namespace object
```

This way, each subparser completely describes the subcommand.